app/services/match_assignment.rb

```diff
@@ -0,0 +1,55 @@
+class MatchAssignment
+  def initialize(user)
+    @user = user
+    @success = false
+  end
+
+  def perform
+    ActiveRecord::Base.transaction do
+      rollback!("user already has match") if user.matches.any?
+
+      @_receiver = available_receiver or rollback!("no receiver available")
```

Comment:
Just one quick question before approving, why the underscore?

Reply:
It's not necessary now. It's usually a pattern to mark it as private when you memoization so it doesn't deceive developers due to the actual method name:

```ruby
def receiver
  @_receiver ||= ...
end
```

However, I opted by not doing memoization here. I left it without actually giving it a second thought. What do you think?

Reply:
Not memoizing so.... kinda not needed I believe, do what you think it's best. It's good both ways for me (prefer not having `_` if not memoizing)